lib/Optimizer/Transforms/QuakeSynthesizer.cpp

```diff
@@ -561,12 +632,21 @@ class QuakeSynthesizer
                   return builder.create<cudaq::cc::CastOp>(
                       loc, cudaq::cc::PointerType::get(stateTy), rawPtr);
                 });
-            continue;
+          } else if (stateData != nullptr && !stateData->isEmpty()) {
+            // Special case of running on a simulator in a different address
+            // space, when we have converted state data.
+            if (failed(synthesizeStateArgument(builder, module, counter,
+                                               argument, *stateData,
+                                               stateDataOffset)))
+              funcOp.emitOpError("synthesis failed for state*");
+            stateDataOffset++;
           } else {
-            funcOp.emitOpError("synthesis: unsupported argument type for "
-                               "remote devices and simulators: state*");
+            // All other cases are not yet supported (i.e. quantum hardware).
```

Comment:
I don't think we can crash out of the compile if we see a pointer. I'm kind of surprised the tests didn't catch this.

There are perfectly legit cases where an argument to the kernel is a pointer and it is ok. It happens when the pointer isn't really required because the argument was already dealt with in another way.

Strange though that this didn't fail in the tests.